src/test/compile-fail/check-static-recursion-foreign.rs

```diff
@@ -29,5 +29,6 @@ extern "C" {
 }
 
 static B: &'static c_int = unsafe { &test_static };
+//~^ ERROR foreign statics cannot be accessed at compile-time
```

Comment:
Wait, this is correct... I think we have to allow it, and instead make `miri` disallow *reads* from them. Sorry about the detour, I didn't realize borrowing `extern static` was already allowed.